src/components/features/language-selector/LanguageSelectorMobile.tsx

```diff
@@ -1,17 +1,24 @@
+'use client';
+
 import { LanguageIcon, CloseIcon } from '@contentful/f36-icons';
-import { useTranslation } from 'next-i18next';
-import { useRouter } from 'next/router';
+import { useCurrentLocale } from 'next-i18n-router/client';
+import { usePathname, useRouter } from 'next/navigation';
 import { useEffect, useState } from 'react';
 import FocusLock from 'react-focus-lock';
+import { useTranslation } from 'react-i18next';
 import { twMerge } from 'tailwind-merge';
 
 import { Portal } from '@src/components/shared/portal';
+import i18nConfig, { locales } from '@src/i18n/config';
 
 export const LanguageSelectorMobile = ({ localeName, displayName }) => {
-  const { locale, locales } = useRouter();
+  const currentLocale = useCurrentLocale(i18nConfig);
+  const pathname = usePathname();
   const router = useRouter();
   const { t } = useTranslation();
   const [showDrawer, setShowDrawer] = useState(false);
+  const pathnameHasLocale = locales.includes(pathname.slice(1, 6));
```

Comment:
Can we add a comment on why 1 and 6.

I assume it's because of the `/en-US` (or other locale) prefix which is 6 chars?